sound/soc/sof/intel/apl.c

```diff
@@ -27,12 +27,19 @@ static const struct snd_sof_debugfs_map apl_dsp_debugfs[] = {
 
 static int apl_register_clients(struct snd_sof_dev *sdev)
 {
-	return hda_probes_register(sdev);
+	int ret;
+
+	ret = hda_probes_register(sdev);
+	if (ret)
+		return ret;
+
+	return hda_dma_trace_register(sdev);
```

Comment:
if the first one succeeds but the second one fails, `apl_unregister_clients()` won't be called, right? Maybe we have to do partial clean up in `*_register_clients()` or call `sof_unregister_clients()` on the error path in `sof_probe_continue()`

Reply:
Yes, I had a note about this, looks like I have missed it badly, thanks!